### PR TITLE
Handle back-ticks in expression

### LINKED ISSUE #3273: Bad parsing when bound data contains a backtick
### Vue.js version
1.0.26

### Reproduction Link
http://codepen.io/sirlancelot/pen/zBPVBo?editors=1010#0

### Steps to reproduce
* Bind an inline object with a value containing the backtick character

### What is Expected?
* Object is reproduced properly in the data model

### What is actually happening?
* `scope.` is being added to everything between the backtick and an escaped quote.

We're using props to curry an initial data model from the server to the browser. We're doing it in a manner similar to the example shown in the CodePen. Basically, our root Vue instance contains a list of props that will show up on the element and are populated by the server.

### PR DESCRIPTION
**Pull request by skyronic**
This fixes #3273 but might introduce other bugs. I'm not sure of it.

### CODE REVIEW

File: src/parsers/expression.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const improperKeywordsRE =
 
 const wsRE = /\s/g
 const newlineRE = /\n/g
-const saveRE = /[\{,]\s*[\w\$_]+\s*:|('(?:[^'\\]|\\.)*'|"(?:[^"\\]|\\.)*"|`(?:[^`\\]|\\.)*\$\{|\}(?:[^`\\]|\\.)*`|`(?:[^`\\]|\\.)*`)|new |typeof |void /g
+const saveRE = /[\{,]\s*[\w\$_]+\s*:|('(?:[^'\\]|\\.)*'|"(?:[^"\\]|\\.)*"|`(?:[^`\\]|\\.)*\$\{|\}(?:[^`\\"']|\\.|)*`|`(?:[^`\\]|\\.)*`)|new |typeof |void /g
```
Comment (skyronic) [inline src/parsers/expression.js]: There's a stray "|" which should be removed here I guess.

**Comment by yyx990803** [comment]
Can you add an accompanying test case?

**Comment by skyronic** [comment]
@yyx990803: updated commit with test case.

**Comment by sirlancelot** [comment]
Thanks!

**Pull request merged**